src/v/rpc/connection_cache.cc

```diff
@@ -57,9 +57,12 @@ ss::future<> connection_cache::remove(model::node_id n) {
 
 /// \brief closes all client connections
 ss::future<> connection_cache::stop() {
-    return parallel_for_each(_cache, [](auto& it) {
-        auto& [_, cli] = it;
-        return cli->stop();
+    return _mutex.with([this]() {
```

Comment:
i don't see how adding a mutex here prevents the case described in the commit message:

> When connection cache is being closed we have to make sure that no other
fiber is in the middle of creating connection as it may result in
resource leak.

if such a race were taking place then it would seem that the same race aligned slightly different in time would allow the connection creation to occur after this mutex is released, leading to the same resource leak? point being, it seems like either the race is more specific and the commit message needs updating, or a flag of some sort needs to be set in stop to prevent subsequent creations from starting?

Reply:
you are right, the commit message is incorrect